File: section03_rn_bridge_android/example_bridge_android/specs/NativeExampleModule.ts
// specs/NativeExampleModule.ts
import type { TurboModule } from 'react-native';
import { TurboModuleRegistry } from 'react-native';

export interface Spec extends TurboModule {
  printMessage(): void;
}

export default TurboModuleRegistry.getEnforcing<Spec>('NativeExampleModule');

File: section03_rn_bridge_android/example_bridge_android/android/app/src/main/java/com/example_bridge_android/NativeExampleModule.kt
package com.example_bridge_android

import android.util.Log
import com.example_bridge_android.NativeExampleModuleSpec
import com.facebook.react.bridge.ReactApplicationContext

class NativeExampleModule(
    reactContext: ReactApplicationContext
) : NativeExampleModuleSpec(reactContext) {

    override fun getName() = NAME

    override fun printMessage() {
        Log.d("NativeExampleModule", "Native message with React Native")
    }

    companion object {
        const val NAME = "NativeExampleModule"
    }
}